From 7a3628327736291697c0c97f5232e229c5802cb8 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 31 Jul 2024 15:47:00 +0200
Subject: [PATCH 1069/1695] dts: stm32: Fix ranges and dma-range property of
 mlahb node on stm32mp21

The parent node of mlahb is the root node which fixes size-cells to 2.
this implies that the bus address must de declared in 64-bit format.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2e4b5a85b0e25d37fe5a67902370bf686cccec75 dts: stm32: Fix ranges and dma-range property of mlahb node on stm32mp21]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I08bca61429fd761a3a9f015ff968bf33e1d89a30
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/396245
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index e5a1c98da..d5da6cbc5 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -120,8 +120,8 @@
 		compatible = "st,mlahb", "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
-		ranges = <0x0 0x0  0xFFFFFFFC>;
-		dma-ranges = <0x0 0x0 0xFFFFFFFC>;
+		ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
+		dma-ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
-- 
2.47.2

